packages/@aws-cdk/aws-servicediscovery/test/test.service.ts

```diff
@@ -0,0 +1,366 @@
+            Name: "http"
+          },
+        },
+        MyServiceA1F951EB: {
+          Type: "AWS::ServiceDiscovery::Service",
+          Properties: {
+            Description: "service description",
+            HealthCheckCustomConfig: {
+              FailureThreshold: 3,
+            },
+            Name: "service",
+            NamespaceId: {
+              "Fn::GetAtt": [
+                "MyNamespaceD0BB8558",
+                "Id"
+              ]
+            }
+          }
+        }
+      }
+    });
+
+    test.done();
+  },
+
+  'Service for DNS namespace'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+
+    const namespace = new servicediscovery.Namespace(stack, 'MyNamespace', {
+      name: 'dns',
+      type: servicediscovery.NamespaceType.DnsPublic,
+    });
+
+    new servicediscovery.Service(stack, 'MyService', {
+      namespace,
+      name: 'service',
+      description: 'service description',
+      healthCheckCustomConfig: {
+        failureThreshold: 3,
+      }
+    });
+
+    // THEN
+    expect(stack).toMatch({
+      Resources: {
+        MyNamespaceD0BB8558: {
+          Type: "AWS::ServiceDiscovery::PublicDnsNamespace",
+          Properties: {
+            Name: "dns"
+          }
+        },
+        MyServiceA1F951EB: {
+          Type: "AWS::ServiceDiscovery::Service",
+          Properties: {
+            Description: "service description",
+            DnsConfig: {
+              DnsRecords: [
+                {
+                  TTL: "60",
+                  Type: "A"
+                }
+              ],
+              NamespaceId: {
+                "Fn::GetAtt": [
+                  "MyNamespaceD0BB8558",
+                  "Id"
+                ]
+              },
+              RoutingPolicy: "MULTIVALUE"
+            },
+            HealthCheckCustomConfig: {
+              FailureThreshold: 3
+            },
+            Name: "service",
+            NamespaceId: {
+              "Fn::GetAtt": [
+                "MyNamespaceD0BB8558",
+                "Id"
+              ]
+            }
+          }
+        }
+      }
+    });
+
+    test.done();
+  },
+
+  'Service for DNS namespace with A and AAAA records'(test: Test) {
+    // GIVEN
+    const stack = new cdk.Stack();
+
+    const namespace = new servicediscovery.Namespace(stack, 'MyNamespace', {
+      name: 'dns',
+      type: servicediscovery.NamespaceType.DnsPublic,
+    });
+
+    new servicediscovery.Service(stack, 'MyService', {
+      namespace,
+      dnsRecordType: servicediscovery.DnsRecordType.A_AAAA
+    });
+
+    // THEN
+    expect(stack).toMatch({
+      Resources: {
+        MyNamespaceD0BB8558: {
+          Type: "AWS::ServiceDiscovery::PublicDnsNamespace",
+          Properties: {
+            Name: "dns"
+          }
+        },
+        MyServiceA1F951EB: {
+          Type: "AWS::ServiceDiscovery::Service",
+          Properties: {
+            DnsConfig: {
+              DnsRecords: [
```

Comment:
Hello,

Author of https://github.com/awslabs/aws-cdk/pull/1714 here (a lot of my code has been copied here).

I'm just wondering how the special case of `A` and `AAAA` records for a service is treated here. It should generate an array of `DnRecords` and it currently only generates one record of type `A, AAAA` which, as far as I know, is not valid CF (https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-properties-servicediscovery-service-dnsrecord.html#cfn-servicediscovery-service-dnsrecord-type)

Reply:
Thanks for bringing this up (and thanks also for your contribution!) I'll look into this and will be adding more integration tests later.